Sources/SwiftInspectorVisitors/NestableTypeInfo.swift

```diff
@@ -22,26 +22,38 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 
-public struct ClassInfo: Codable, Equatable {
+protocol NestableTypeInfo {
+
+  init(
```

Comment:
We need it for [this line](https://github.com/fdiaz/SwiftInspector/pull/78/files#diff-1be30608ec9e31cb017d12cd2d0e0d278bf39f9c6ad8bee10bf7a25baaa3a798R130). I agree it's weird. Maybe I could define a static method that returned `Self` rather than an `init`? Curious to hear your thoughts @fdiaz. I built it this way in order to ensure we didn't have unreachable `else` statements 😅

Reply:
turns out you had a great idea here https://github.com/fdiaz/SwiftInspector/pull/78#discussion_r578060743